query/expression.go

```diff
@@ -23,6 +23,55 @@ import (
 	"github.com/square/metrics/query/aggregate"
 )
 
+const simultaneousFetchLimit = 4
```

Comment:
I'm not sure if it's safe to move this as a parameter into the evaluation context, since it's possible for many expressions to be evaluating simultaneously. (It's still pretty safe, just slightly less so, taking 4 times fewer concurrent expressions than the solution without threading).